cmd/config/set.go

```diff
@@ -86,6 +88,43 @@ func runSet(cmd *cobra.Command, vp *viper.Viper, key, value string) error {
 	if err != nil {
 		return fmt.Errorf("cannot assign value=%s for key=%s, expected type=%s: %w", value, key, typ, err)
 	}
-	vp.Set(key, newVal)
-	return vp.WriteConfig()
+
+	// Create a file only viper from the configured file to avoid writing
```

Comment:
nit (had difficulty parsing this the first time around):
```suggestion
	// Create a file-only viper config from the configured file to avoid writing
```